### PR TITLE
Make share tools work in shadow dom.

### PR DESCRIPTION
**Pull request by felixcs1**
Jira ticket: https://jira.dev.bbc.co.uk/browse/NEWSSPEC-21601?page=com.atlassian.jira.plugin.system.issuetabpanels%3Aall-tabpanel

-Changed holderEl to be a dom element instead of a string.
-Updated the tests to pass

### CODE REVIEW

File: _SpecRunner.html [hunk 1]
```diff
@@ -0,0 +1,95 @@
+<!doctype html>
```
Comment (felixcs1) [inline _SpecRunner.html]: This file presumably turned up from updating grunt-contrib-jasmine from 1.0.0 => 1.0.3

Comment (felixcs1) [reply _SpecRunner.html]: very oddly spaced!

Comment (ChrisBAshton) [reply _SpecRunner.html]: Yeah! We should remove this and add it to .gitignore: https://github.com/bbc/news-vj-sharetools/blob/master/.gitignore

File: tests/controllerSpec.js [hunk 2]
```diff
@@ -3,9 +3,12 @@ define(['ShareTools'], function (ShareToolsController) {
     var controller;
 
     beforeEach(function() {
+        holderEl = document.createElement('div');
```
Comment (ChrisBAshton) [inline tests/controllerSpec.js]: put a `var` here.

File: tests/controllerSpec.js [hunk 3]
```diff
@@ -53,13 +56,14 @@ define(['ShareTools'], function (ShareToolsController) {
         });
     });
 
+    
     describe('ShareTools Controller', function () {
-
+
```
Comment (ChrisBAshton) [inline tests/controllerSpec.js]: Remove indent

File: tests/controllerSpec.js [hunk 4]
```diff
@@ -53,13 +56,14 @@ define(['ShareTools'], function (ShareToolsController) {
...
+    
     describe('ShareTools Controller', function () {
-
+        
         it('should construct the right share url for predefined networks', function () {
             var fbShareUrl = controller.getShareTargetUrl('facebook');
             expect(fbShareUrl).toEqual('https://www.facebook.com/dialog/share?app_id=58567469885&redirect_uri=http%3A%2F%2Fwww.bbc.co.uk%2Fnews%2Fspecial%2Fshared%2Fvj_sharetools%2Ffb_red_uri.html%3Fst_cb%3Dfacebook%23state%3Dfeed&display=popup&locale=en_GB&href=http%3A%2F%2Fwww.bbc.co.uk&quote=Facebook%20share%20message');
         });
-
+
```
Comment (ChrisBAshton) [inline tests/controllerSpec.js]: Remove Indent

**Review by ChrisBAshton — COMMENTED** [review]
Looks promising!

**Review by ChrisBAshton — APPROVED** [review]
This looks great - thanks Felix!

Please merge, then `npm version minor` and `git push && git push --tags`. Also then go into GitHub and 'publish' the release, linking to this PR

**Pull request merged**